### PR TITLE
Load Configuration from Meta Master for all clients

### PR DESCRIPTION
**Pull request by bradyoo**
The InetSocketAddress for configuration must always be the master.

While the address could be the master or the job master.

@calvinjia

### CODE REVIEW

File: job/client/src/main/java/alluxio/worker/job/RetryHandlingJobMasterClient.java [hunk 1]
```diff
@@ -60,12 +60,6 @@ protected long getServiceVersion() {
     return Constants.JOB_MASTER_WORKER_SERVICE_VERSION;
   }
 
-  @Override
```
Comment (bradyoo) [inline job/client/src/main/java/alluxio/worker/job/RetryHandlingJobMasterClient.java]: These lines were added to get around not being able to grab configuration in beforeConnect() in the AbstractClient.java

File: job/client/src/main/java/alluxio/client/job/RetryHandlingJobMasterClient.java [hunk 2]
```diff
@@ -65,13 +65,6 @@ protected long getServiceVersion() {
     return Constants.JOB_MASTER_CLIENT_SERVICE_VERSION;
   }
 
-  @Override
```
Comment (bradyoo) [inline job/client/src/main/java/alluxio/client/job/RetryHandlingJobMasterClient.java]: See https://github.com/Alluxio/alluxio/pull/10069/files#r332794028

File: core/common/src/main/java/alluxio/AbstractClient.java [hunk 3]
```diff
@@ -56,6 +56,9 @@
 
   protected InetSocketAddress mAddress;
 
+  /** Address to load configuration. */
```
Comment (gpang) [inline core/common/src/main/java/alluxio/AbstractClient.java]:
```suggestion
  /** Address to load configuration, which may differ from {@code mAddress}. */
```

File: core/common/src/main/java/alluxio/AbstractMasterClient.java [hunk 4]
```diff
@@ -29,6 +29,9 @@
   /** Client for determining the master RPC address. */
   private final MasterInquireClient mMasterInquireClient;
 
+  /** Client for determining the RPC address for getting configuration. */
```
Comment (gpang) [inline core/common/src/main/java/alluxio/AbstractMasterClient.java]:
```suggestion
  /** Client for determining configuration RPC address, which may differ from target address. */
```

File: job/client/src/main/java/alluxio/client/job/RetryHandlingJobMasterClient.java [hunk 5]
```diff
@@ -65,13 +65,6 @@ protected long getServiceVersion() {
...
-  protected void beforeConnect()
-      throws IOException {
-    // Job master client does not load cluster-default configuration because only the master
```
[the remark this answers is not in the document]
Comment (bradyoo) [reply job/client/src/main/java/alluxio/client/job/RetryHandlingJobMasterClient.java]: For the client, the comment will be no longer true as there will be report metrics that will be using the RetryHandlingJobMasterClient through the cli. 

For the worker one, the comment may be still True but configuration should still be fetch-able.

**Comment by bradyoo** [comment]
Feedback addressed.

**Comment by bradyoo** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by bradyoo** [comment]
alluxio-bot, merge this please.

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**